main.go

```diff
@@ -176,6 +177,44 @@ func locateDotGit(dir string) (string, error) {
 	return "", fmt.Errorf("Couldn't find .git. Maybe you are not inside of a git repo")
 }
 
+func getURLAliases() (map[string]string, error) {
+	usr, err := user.Current()
+	if err != nil {
+		return map[string]string{}, err
+	}
+
+	path := path.Join(usr.HomeDir, ".gitconfig")
+
+	cfg, err := ini.Load(path)
+	if err != nil {
+		return map[string]string{}, err
+	}
+
+	sections := cfg.Sections()
+	aliases := map[string]string{}
+
+	for _, elem := range sections {
+		name := elem.Name()
+		isURLSection := strings.Contains(name, "url")
+		hasHostName := strings.Contains(name, "github") || strings.Contains(name, "gitlab")
+
+		if isURLSection && hasHostName {
+			section := cfg.Section(name)
+			key, err := section.GetKey("insteadOf")
+			if err != nil {
+				return map[string]string{}, err
+			}
+
+			regex := regexp.MustCompile("[-\\w]+@(github.com|gitlab.com)[^\"]+")
+			resolvedAlias := regex.FindString(name)
```

Comment:
If we include `url` into the regular expression I'm pretty sure we will be able to get rid of `isURLSection` and `hasHostName`. 

Can we basically use regular expression to do both things at once: check if the section is the right one by matching the regexp and resolve the URL by extracting it from the capture groups of the regexp?

Reply:
I think I made this the way you wanted